### PR TITLE
Automatic query submission #LMR-972

### PR DESCRIPTION
**Pull request by kubedo8**

### CODE REVIEW

File: src/app/shared/search-box/search-box.component.ts [hunk 1]
```diff
@@ -127,18 +129,42 @@ export class SearchBoxComponent implements OnInit, OnDestroy {
 
   public onAddQueryItem(queryItem: QueryItem) {
     this.queryItems.push(queryItem);
-    this.queryItemsControl.push(queryItemToForm(queryItem));
+
+    const queryItemForm = queryItemToForm(queryItem);
+    this.queryItemsControl.push(queryItemForm);
+
+    if (queryItemForm.valid) {
```
Comment (livthomas) [inline src/app/shared/search-box/search-box.component.ts]: If I understand it correctly, `queryItemForm` represents a form group responsible for just a single query item. What happens if you already have an invalid query item in the search box and add a new one which is valid? This condition will be true but the form should not be submitted.

File: src/app/shared/search-box/search-box.component.ts [hunk 2]
```diff
@@ -127,18 +129,42 @@ export class SearchBoxComponent implements OnInit, OnDestroy {
...
+    }
   }
 
   public onRemoveQueryItem(index: number) {
     this.queryItems.splice(index, 1);
+
+    const isQueryItemValid =  this.queryItemsControl.at(index).valid;
     this.queryItemsControl.removeAt(index);
+
+    if(isQueryItemValid) {
```
Comment (livthomas) [inline src/app/shared/search-box/search-box.component.ts]: Why do you check the validity of the query item which is being removed? Does it matter if it was valid or not? I think more important information is if the whole search box form is valid at this point.

File: src/app/shared/search-box/search-box.component.ts (hunk 1 above)
Comment (kubedo8) [reply src/app/shared/search-box/search-box.component.ts]: Good point! I will definitely change it :)

File: src/app/shared/search-box/search-box.component.ts (hunk 2 above)
Comment (kubedo8) [reply src/app/shared/search-box/search-box.component.ts]: It is same as above.. thanks for finding that 👍

**Pull request merged**